internal/cmd/io_binary.go

```diff
@@ -0,0 +1,236 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"os/exec"
+	"sync"
+	"time"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/pkg/errors"
+)
+
+const (
+	pipeRootBinary       = `\\.\pipe`
```

Comment:
I feel this is better suited with the format that we're expecting the pipe to be named with. 
```go
const binaryPipeFmt = `\\.\pipe\binary-%s-stdout`
```

and then you can simplify the later call. 
```go
stdoutPipe := fmt.Sprintf(binaryPipefmt, id)
```